git-branchless-navigation/src/lib.rs

```diff
@@ -544,6 +550,36 @@ pub fn switch(
                 None => return Ok(Err(ExitCode(1))),
             }
         }
+        Target::Revset(target) => {
+            let commit_sets = resolve_commits(
+                effects,
+                &repo,
+                &mut dag,
+                &[target.clone()],
+                &ResolveRevsetOptions::default(),
+            )?;
+
+            let commit_set = union_all(&commit_sets);
+            let commit_set = dag.query_heads(commit_set)?;
+            let commits = sorted_commit_set(&repo, &dag, &commit_set)?;
+
+            match commits.as_slice() {
+                [commit] => Some(CheckoutTarget::Unknown(commit.get_oid().to_string())),
+                [] => None,
+                [..] => {
+                    writeln!(
+                        effects.get_error_stream(),
+                        "Cannot switch to target: expected '{target}' to contain 1 head, but found {}.",
```

Comment:
comment (non-blocking): In principle we could list all of the candidate targets, which we also do when `git next` or `git prev` is ambiguous, but it's not necessary to implement that here.

Reply:
Yes, that makes sense, but I'm going to defer on that for now and leave that as a great future enhancement.